pages/dao/[symbol]/proposal/components/instructions/UXDProtocol/EditController.tsx

```diff
@@ -0,0 +1,151 @@
+    setFormErrors({})
+    setForm({ ...form, [propertyName]: value })
+  }
+
+  const validateInstruction = async (): Promise<boolean> => {
+    const { isValid, validationErrors } = await isFormValid(schema, form)
+    setFormErrors(validationErrors)
+    return isValid
+  }
+
+  async function getInstruction(): Promise<UiInstruction> {
+    const isValid = await validateInstruction()
+
+    if (
+      !connection ||
+      !isValid ||
+      !form.governedAccount?.governance?.account.governedAccount
+    ) {
+      return {
+        serializedInstruction: '',
+        isValid: false,
+        governance: form.governedAccount?.governance,
+      }
+    }
+    const uxdProgramId =
+      form.governedAccount?.governance?.account.governedAccount
+    const client = uxdClient(uxdProgramId)
+    const authority = form.governedAccount.governance.pubkey
+    const ix = client.createEditControllerInstruction(
+      new Controller('UXD', UXD_DECIMALS, uxdProgramId),
+      authority,
+      {
+        redeemableGlobalSupplyCap: form.decimalRedeemableGlobalSupplyCap,
+      },
+      { preflightCommitment: 'processed', commitment: 'processed' }
+    )
+
+    return {
+      serializedInstruction: serializeInstructionToBase64(ix),
+      isValid: true,
+      governance: form.governedAccount.governance,
+      shouldSplitIntoSeparateTxs: true,
+    }
+  }
+
+  useEffect(() => {
+    handleSetInstructions(
+      {
+        governedAccount: form.governedAccount?.governance,
+        getInstruction,
+      },
+      index
+    )
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- TODO please fix, it can cause difficult bugs. You might wanna check out https://bobbyhadz.com/blog/react-hooks-exhaustive-deps for info. -@asktree
+  }, [form])
+
+  return (
+    <>
+      <GovernedAccountSelect
+        label="Governed account"
+        governedAccounts={assetAccounts}
+        onChange={(value) => {
+          handleSetForm({ value, propertyName: 'governedAccount' })
+        }}
+        value={form.governedAccount}
+        error={formErrors['governedAccount']}
+        shouldBeGoverned={shouldBeGoverned}
+        governance={governance}
+      />
+      <h5>Redeemable Global Supply Cap</h5>
+      <Switch
+        checked={redeemableGlobalSupplyCapChange}
+        onChange={(checked) => setRedeemableGlobalSupplyCapChange(checked)}
+      />
+      {redeemableGlobalSupplyCapChange ? (
+        <Input
+          label="Redeemable Global Supply Cap"
+          value={form.decimalRedeemableGlobalSupplyCap}
+          type="number"
+          min={0}
+          max={10 ** 12}
+          onChange={(evt) =>
+            handleSetForm({
+              value: evt.target.value,
+              propertyName: 'decimalRedeemableGlobalSupplyCap',
+            })
+          }
+          error={formErrors['decimalRedeemableGlobalSupplyCap']}
+        />
+      ) : null}
+    </>
+  )
+}
+
+export default EditControllerDepository
```

Comment:
nit, controller is not a depository

```suggestion
export default EditController
```